pocs/utils/google/storage.py

```diff
@@ -97,3 +99,247 @@ def upload(self, local_path, remote_path=None):
+                default False.
+
+        Returns:
+            google.cloud.storage.blob.Blob: Blob object.
+        """
+        objs = list()
+        for blob in self.bucket.list_blobs(prefix=prefix):
+            if 'pointing' in blob.name and not include_pointing:
+                continue
+            elif filter_ext is not None and blob.name.endswith(filter_ext) is False:
+                continue
+            else:
+                objs.append(blob)
+
+        return sorted(objs, key=lambda x: x.name)
+
+    def download_file(self, remote_path, save_dir='.', force=False, callback=None):
+        """Downloads (and uncompresses) the image blob data.
+
+        Args:
+            remote_path (str|`google.cloud.storage.blob.Blob`): Blob or path to remote blob.
+                If just the blob name is given then file will be downloaded.
+            save_dir (str, optional): Path for saved file, defaults to current directory.
+            force (bool, optional): Force a download even if file exists locally, default False.
+            callback (callable, optional): A callable object that gets called at end of
+                function.
+
+        Returns:
+            str: Path to local (uncompressed) FITS file
+        """
+        # Get blob object if just a path
+        if isinstance(remote_path, str):
+            remote_path = self.get_file_blob(remote_path)
+
+        output_path = os.path.join(
+            save_dir,
+            remote_path.name.replace('/', '_')
+        )
+
+        # Make dir if needed
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+
+        # If we don't  have file (or force=True) then download directly to
+        # (compressed) FITS file.
+        if not os.path.exists(output_path) or force:
+            with open(output_path, 'wb') as f:
+                remote_path.download_to_file(f)
+
+        # Wait for download
+        timeout = 10
+        while os.path.exists(output_path) is False:
+            timeout -= 1
+            if timeout < 0:
+                return None
+            time.sleep(1)
+
+        # User supplied callback
+        if callback is not None:
+            try:
+                callback()
+            except TypeError:
+                warn('callback must be callable')
+
+        return output_path
+
+    def lookup_fits_header(self, remote_blob, parse_line=None):
+        """Read the FITS header from storage.
+
+        FITS Header Units are stored in blocks of 2880 bytes consisting of 36 lines
+        that are 80 bytes long each. The Header Unit always ends with the single
+        word 'END' on a line (not necessarily line 36).
+
+        Here the header is streamed from Storage until the 'END' is found, with
+        each line given minimal parsing.
+
+        See https://fits.gsfc.nasa.gov/fits_primer.html for overview of FITS format.
+
+        Args:
+            remote_blob (google.cloud.storage.blob.Blob): Blob object
```

Comment:
Any reason to not allow a string as done in the prior method?

Reply:
Good call, thanks.